### PR TITLE
Linting

### PR DESCRIPTION
**Pull request by JakeHartnell**
Adds linting CI and fixes linting errors.

### CODE REVIEW

**Comment by JakeHartnell** [comment]
Could use some feedback on this. Close #109.

There are some outstanding errors I'm not sure what to do with:
```
cmd/staked/testnet.go:300:2: `cdc` can be `github.com/cosmos/cosmos-sdk/codec.JSONMarshaler` (interfacer)
        cdc *codec.Codec, mbm module.BasicManager, chainID, stakeDenom, unbondingPeriod string,
        ^
cmd/staked/genaccounts.go:55:9: shadow: declaration of "err" shadows declaration at line 51 (govet)
                                kb, err := keyring.New(
                                    ^
x/curating/keeper/post.go:4:2: G501: Blacklisted import `crypto/md5`: weak cryptographic primitive (gosec)
        "crypto/md5"
        ^
x/curating/keeper/post.go:158:7: G401: Use of weak cryptographic primitive (gosec)
        h := md5.New()
             ^
x/curating/keeper/post.go:116:12: composites: `github.com/public-awesome/stakebird/x/curating/types.VPPair` composite literal uses unkeyed fields (govet)
        vpPair := types.VPPair{vendorID, postID}
                  ^
x/curating/keeper/post.go:150:37: composites: `github.com/public-awesome/stakebird/x/curating/types.VPPairs` composite literal uses unkeyed fields (govet)
        bz := k.cdc.MustMarshalBinaryBare(&types.VPPairs{vps})
```

Should we be using something else besides MD5? The linter brings up a good point. :sweat_smile:

File: app/app.go [hunk 1]
```diff
@@ -210,7 +210,7 @@ func NewStakebirdApp(
 ) *StakebirdApp {
 
 	// TODO: Remove cdc in favor of appCodec once all modules are migrated.
-	cdc := std.MakeCodec(ModuleBasics)
+	cdc := std.MakeCodec(ModuleBasics) // nolint
```
Comment (JakeHartnell) [inline app/app.go:213]: Went with `nolint` here because there is a TODO above.

File: x/curating/client/rest/query.go [hunk 2]
```diff
@@ -28,9 +28,11 @@ func registerQueryRoutes(cliCtx context.CLIContext, r *mux.Router) {
 	).Methods("GET")
 }
 
+var ok bool
```
Comment (jhernandezb) [inline x/curating/client/rest/query.go:31]: I dont think this is the right fix here , what was the linting issue?

Comment (jhernandezb) [reply x/curating/client/rest/query.go:31]: I think you need to move it inside the methods

```
var ok bool
cliCtx, ok = rest.ParseQueryHeightOrReturnBadRequest(w, cliCtx, r)
```

**Pull request closed without merging**